tools/sgx-lkl-docker

```diff
@@ -44,22 +44,46 @@ function build() {
 FROM ubuntu:18.04
 RUN apt-get update && apt-get install -y ca-certificates
 
-FROM scratch
 # ca-certificates.crt is read by libcurl which is used by the Azure DCAP Client.
-COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
-COPY . /
 
-# Note: WORKDIR also creates the directory.
-WORKDIR /tmp
-ENV TMPDIR=/tmp
+RUN groupadd -g 999 sgxlkl && \
+    useradd -g 999 -u 999 sgxlkl
+
+RUN mkdir /prep && \
+    cd /prep && \
+    mkdir etc && \
+    echo "sgxlkl:x:999:999::/tmp:/usr/sbin/nologin" > etc/passwd && \
+    echo "sgxlkl:x:999:" > etc/group && \
+    mkdir -p home/sgxlkl && \
+    chown sgxlkl:sgxlkl home/sgxlkl && \
+    ls -al home/sgxlkl && \
+    mkdir tmp && \
+    chmod 777 tmp && \
+    mkdir -p etc/ssl/certs && \
+    cp /etc/ssl/certs/ca-certificates.crt etc/ssl/certs && \
+    ls -al etc/ssl/certs
+
+COPY . /prep/home/sgxlkl
+
+# Allows container to be run with arbitrary user while supporting writable disk images.
+RUN chmod -R 777 /prep/home/sgxlkl
 
+FROM scratch
+
+ENV HOME=/home/sgxlkl
+ENV TMPDIR=/tmp
 # See https://github.com/microsoft/Azure-DCAP-Client.
-ENV AZDCAP_CACHE=${TMPDIR}
+ENV AZDCAP_CACHE=\${TMPDIR}
+
+COPY --from=0 /prep /
+
+USER sgxlkl
+WORKDIR \${HOME}
 
 ENTRYPOINT ["/opt/sgx-lkl/lib/external/loader", "/opt/sgx-lkl/bin/sgx-lkl-run-oe",\
```

Comment:
What is "/opt/sgx-lkl/lib/external/loader"?

Reply:
It's a symlink to the dynamic loader (`/opt/sgx-lkl/lib/external/`ld-linux-x86-64.so.2`). If you don't use the loader to run the executable then it would use the absolute loader path embedded in the executable and that would prevent us from mountain into a fixed location at `/opt/sgx-lkl`.

Reply:
Perhaps instead of `loader`, it should have a name that describes more what it is. How about `ld-linux-x86-64`?

Reply:
This is not part of this PR, feel free to open an issue if you think it should be changed.

Reply:
It would be trivial for you to change it here :-)

Reply:
You won.

Reply:
It's not about winning -- it's all about making the project better ;-)